docs/src/test/scala/docs/http/scaladsl/server/TestKitWithActorSpec.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package docs.http.scaladsl.server
+
+//#testkit-actor-integration
+import scala.concurrent.duration._
+import scala.util.{ Failure, Success }
+
+import akka.actor.testkit.typed.scaladsl.TestProbe
+import akka.actor.typed.{ ActorRef, Scheduler }
+import akka.actor.typed.scaladsl.AskPattern._
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import akka.util.Timeout
+
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+object RouteUnderTest {
+  case class Ping(replyTo: ActorRef[String])
+
+  // Your route under test, scheduler is only needed as ask is used
+  def route(someActor: ActorRef[Ping])(implicit scheduler: Scheduler, timeout: Timeout) = get {
+    path("ping") {
+      onComplete(someActor ? Ping) {
+        case Success(s) => complete(s)
+        case Failure(t) => complete(t.toString)
+      }
```

Comment:
Same here onSuccess? or even just complete with the magic future marshaller?

Reply:
good idea